File: src/main/java/net/atos/client/brp/util/FieldPropertyVisibilityStrategy.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.brp.util;

import java.lang.reflect.Field;
import java.lang.reflect.Method;

import jakarta.json.bind.config.PropertyVisibilityStrategy;

public class FieldPropertyVisibilityStrategy implements PropertyVisibilityStrategy {

    @Override
    public boolean isVisible(final Field field) {
        return true;
    }

    @Override
    public boolean isVisible(final Method method) {
        return false;
    }
}

File: src/main/kotlin/net/atos/client/brp/util/FieldPropertyVisibilityStrategy.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.brp.util

import jakarta.json.bind.config.PropertyVisibilityStrategy
import java.lang.reflect.Field
import java.lang.reflect.Method

class FieldPropertyVisibilityStrategy : PropertyVisibilityStrategy {
    override fun isVisible(field: Field) = true

    override fun isVisible(method: Method) = false
}
